### PR TITLE
Support H2RBox-v2

### PR DESCRIPTION
**Pull request by yuyi1005**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support H2RBox-v2.

## Modification

Support H2RBox-v2.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/h2rbox_v2/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# H2RBox-v2
```
Comment (RangiLyu) [inline configs/h2rbox_v2/README.md:1]: Update the Model Zoo in readme https://github.com/open-mmlab/mmrotate/tree/1.x#model-zoo

File: configs/h2rbox_v2/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+> [H2RBox-v2: Boosting HBox-supervised Oriented Object Detection via Symmetric Learning](https://arxiv.org/pdf/2304.04403)
+
+<!-- [ALGORITHM] -->
+
```
Comment (yangxue0827) [inline configs/h2rbox_v2/README.md:6]: Figure of pipeline. Send image to @zytx121

**Review by zytx121 — DISMISSED** [review]
LGTM. I will fix the CI after this PR is merged.

File: configs/h2rbox_v2/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+| ResNet50 (1024,1024,200) | 72.59 |   1x    |  10.10   |      29.1      |   -   |     2      |       [h2rbox_v2-le90_r50_fpn-1x_dota](./h2rbox_v2-le90_r50_fpn-1x_dota.py)       |         [model](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn-1x_dota/h2rbox_v2-le90_r50_fpn-1x_dota-fa5ad1d2.pth)   \| [log](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn-1x_dota/h2rbox_v2-le90_r50_fpn-1x_dota-20230313_103051.json)         |
+| ResNet50 (1024,1024,200) | 78.25 |   1x    |  10.33   |      29.1      | MS+RR |     2      | [h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota](./h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota.py) | [model](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota/h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota-5e0e53e1.pth) \| [log](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn-1x_dota/h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota-20230324_011934.json) |
```
Comment (zytx121) [inline configs/h2rbox_v2/README.md:22]:
```suggestion
| ResNet50 (1024,1024,200) | 78.25 |   1x    |  10.33   |      29.1      | MS+RR |     2      | [h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota](./h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota.py) | [model](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota/h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota-5e0e53e1.pth) \| [log](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota/h2rbox_v2-le90_r50_fpn_ms_rr-1x_dota-20230324_011934.json) |
```

File: configs/h2rbox_v2/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+| :----------------------: | :---: | :-----: | :------: | :------------: | :-: | :--------: | :-------------------------------------------------------------------------: | :-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+| ResNet50 (1024,1024,200) | 64.76 |   1x    |  10.95   |      29.1      |  -  |     2      | [h2rbox_v2-le90_r50_fpn-1x_dotav15](./h2rbox_v2-le90_r50_fpn-1x_dotav15.py) | [model](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn-1x_dotav15/h2rbox_v2-le90_r50_fpn-1x_dotav15-3adc0309.pth) \| [log](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn-1x_dota/h2rbox_v2-le90_r50_fpn-1x_dotav15-20230316_192940.json) |
```
Comment (zytx121) [inline configs/h2rbox_v2/README.md:28]:
```suggestion
| ResNet50 (1024,1024,200) | 64.76 |   1x    |  10.95   |      29.1      |  -  |     2      | [h2rbox_v2-le90_r50_fpn-1x_dotav15](./h2rbox_v2-le90_r50_fpn-1x_dotav15.py) | [model](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn-1x_dotav15/h2rbox_v2-le90_r50_fpn-1x_dotav15-3adc0309.pth) \| [log](https://download.openmmlab.com/mmrotate/v1.0/h2rbox_v2/h2rbox_v2-le90_r50_fpn-1x_dotav15/h2rbox_v2-le90_r50_fpn-1x_dotav15-20230316_192940.json) |
```

**Review by yangxue0827 — COMMENTED** [review]
LGTM

**Review by RangiLyu — APPROVED** [review]

**Comment by Petros626** [comment]
@yangxue0827 I can't find the source files for `H2RBox-v2` in `mmrotate`.

**Pull request merged**